### PR TITLE
Ft multiple partition support

### PR DESCRIPTION
**Pull request by mmichailidis**
The second part of the merged PR from spring-cloud-stream ([this](https://github.com/spring-cloud/spring-cloud-stream/pull/1836)). 

Cannot think of a way to test this using the currently provided tests to check the queues created

### CODE REVIEW

**Comment by garyrussell** [comment]
Thanks, but we need some test coverage.

File: spring-cloud-stream-binder-rabbit-core/src/main/java/org/springframework/cloud/stream/binder/rabbit/provisioning/RabbitExchangeQueueProvisioner.java [hunk 1]
```diff
@@ -170,8 +174,26 @@ public ConsumerDestination provisionConsumerDestination(String name, String grou
...
-					.map(destination -> doProvisionConsumerDestination(destination, group,
-							properties).getName())
+					.flatMap(destination -> {
+						if (properties.isPartitioned() && !ObjectUtils.isEmpty(properties.getInstanceIndexList())) {
+							List<String> consumerDestinationNames = new ArrayList<>();
+
+							for (Integer index : properties.getInstanceIndexList()) {
+								ExtendedConsumerProperties<RabbitConsumerProperties> temporaryProperties =
+										new ExtendedConsumerProperties<>(properties.getExtension());
+								BeanUtils.copyProperties(properties, temporaryProperties);
```
Comment (garyrussell) [inline spring-cloud-stream-binder-rabbit-core/src/main/java/org/springframework/cloud/stream/binder/rabbit/provisioning/RabbitExchangeQueueProvisioner.java]: We can do this just once, outside the loop.

Comment (mmichailidis) [reply spring-cloud-stream-binder-rabbit-core/src/main/java/org/springframework/cloud/stream/binder/rabbit/provisioning/RabbitExchangeQueueProvisioner.java]: Wanted to be sure there will be no references between objects in-case of something parallel implemented in the future. If that thought is invalid then I can move it outside yes.

Comment (garyrussell) [reply spring-cloud-stream-binder-rabbit-core/src/main/java/org/springframework/cloud/stream/binder/rabbit/provisioning/RabbitExchangeQueueProvisioner.java]: No that's fine; this is only during provisioning so it's not really important.

Merging.

**Pull request merged**